src/platform/Linux/bluez/ChipDeviceScanner.cpp

```diff
@@ -273,6 +277,24 @@ int ChipDeviceScanner::MainLoopStartScan(ChipDeviceScanner * self)
         self->RemoveDevice(bluez_object_get_device1(&object));
     }
 
+    /* Search for matter/chip UUID only */
+    g_variant_builder_init(&uuidsBuilder, G_VARIANT_TYPE("as"));
+    g_variant_builder_add(&uuidsBuilder, "s", CHIP_BLE_UUID_SERVICE_STRING);
+    uuids = g_variant_builder_end(&uuidsBuilder);
```

Comment:
how are these freed? I would expect some g_variant_unref at least on error cases (or is this not required because add takes ownership of  uuids and set_filter will  take ownership of filter in case of failure?)

Please move the declarations to minimize scope instead of using C-style 'at top of function', like:

```cpp
GVarianBuilder uuidsBuilder;
g_variant_builder_init(&uuidsBuilder, G_VARIANT_TYPE("as"));
g_variant_builder_add(&uuidsBuilder, "s", CHIP_BLE_UUID_SERVICE_STRING);
GVariant *￼uuids = g_variant_builder_end(&uuidsBuilder);

```

Reply:
g_dbus_proxy_call_sync "consumes" the GVariant (takes the ownership). "g_atomic_ref_count_dec: assertion 'g_atomic_int_get (arc) > 0' failed" message is triggered if you try to unref "filter".

Even for failure scenario, parsing "filter" is the first operation before accepting the command.

I will fix declarations and re-send the PR.